jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -201,6 +217,43 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 		return notifications;
 	}
 
+	protected static int getDeadlineRepeatLimit(String deadlineStr) {
+		String[] repeatableParts = DateTimeUtils.parseISORepeatable(deadlineStr);
+		String repeatLimit = repeatableParts[0];
+		if (!repeatLimit.isEmpty()) {
+			return Integer.parseInt(repeatLimit);
+		} else {
+			return -1;
+		}
+	}
+
+	protected static Date getDeadlineDurationDate(String durationStr) {
+		// handles iso repetable and both period and date notation
+		try {
+
+			if (DateTimeUtils.isRepeatable(durationStr)) {
+				String[] repeatableParts = DateTimeUtils.parseISORepeatable(durationStr);
+				String tempTimeDelay = repeatableParts[1];
+				String tempDateTimeStr = repeatableParts[2];
+
+				if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
+					return new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + Duration.parse(tempDateTimeStr).toMillis());
+				} else {
+					return new Date(DateTimeUtils.parseDateTime(tempTimeDelay)+ DateTimeUtils.parseDateAsDuration(tempDateTimeStr));
```

Comment:
This line won't work when there is `R/<duration>/<endTime>` which is also a valid use case (parseDateTime cannot parse duration). But since in case when the third part is date you don't care about the second part (whether it is duration or startTime), you can just do:
You can make it more precise by doing `OffsetDateTime.parse(tempDateTimeStr, DateTimeFormatter.ISO_DATE_TIME);` and then calling `parsedOffsetDateTime.toInstant().toEpochMilli()` which you can use in Date constructor. Right now we are parsing the middle part and then adding duration to it, which is not needed at all.

Reply:
agreed. there is a method DateTimeUtils.parseDateTime which does that (and works for us in case of repetable). ill just use it

Reply:
First IF handles `R/startDate/duration`
And the second IF can handle both `R/startDate/endDate` and `R/duration/endDate` since with deadlines you are only interested in `endDate`. So I would just parse the `tempDateTimeStr` as ISO Date since you "don't care" about the middle part at all.

Reply:
i updated the code just now there are checks for both and added 2 tests